app/v1/connectors/DesConnector.scala

```diff
@@ -17,47 +17,59 @@
 package v1.connectors
 
 import config.AppConfig
-import javax.inject.{Inject, Singleton}
+import javax.inject.{ Inject, Singleton }
 import play.api.Logger
 import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.http.logging.Authorization
 import uk.gov.hmrc.play.bootstrap.http.HttpClient
-import v1.connectors.httpparsers.StandardDesHttpParser
-import v1.models.des.{AmendBFLossResponse, CreateBFLossResponse}
-import v1.models.domain.{AmendBFLoss, BFLoss}
-import v1.models.requestData.{AmendBFLossRequest, CreateBFLossRequest}
+import v1.connectors.httpparsers.StandardDesHttpParser._
+import v1.models.des.{ AmendBFLossResponse, CreateBFLossResponse }
+import v1.models.domain.{ AmendBFLoss, BFLoss }
+import v1.models.requestData.{ AmendBFLossRequest, CreateBFLossRequest, DeleteBFLossRequest }
 
-import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.{ ExecutionContext, Future }
 
 @Singleton
 class DesConnector @Inject()(http: HttpClient, appConfig: AppConfig) {
 
   val logger = Logger(this.getClass)
 
-  private[connectors] def desHeaderCarrier(implicit hc: HeaderCarrier): HeaderCarrier =
+  private def desHeaderCarrier(implicit hc: HeaderCarrier): HeaderCarrier =
     hc.copy(authorization = Some(Authorization(s"Bearer ${appConfig.desToken}")))
       .withExtraHeaders("Environment" -> appConfig.desEnv)
 
   def createBFLoss(createBFLossRequest: CreateBFLossRequest)(implicit hc: HeaderCarrier,
-                                   ec: ExecutionContext): Future[DesOutcome[CreateBFLossResponse]] = {
+                                                             ec: ExecutionContext): Future[DesOutcome[CreateBFLossResponse]] = {
 
-    val nino    = createBFLossRequest.nino.nino
+    val nino = createBFLossRequest.nino.nino
 
-    val url = s"${appConfig.desBaseUrl}/income-tax/brought-forward-losses/$nino"
+    def doIt(implicit hc: HeaderCarrier) =
```

Comment:
Why was this moved into its own function?

Reply:
So that we can pass an explicit HeaderCarrier without the need to explicitly supply all of the other implicits every time.